nautobot/docs/plugins/templates.md

```diff
@@ -0,0 +1,106 @@
+# Templates for Plugins
+
+Nautobot comes with a variety of plugins that allow for a lot of flexibility
+while keeping the page style consistent with the rest of the application.
+This document presents these templates and their features.
+
+You can use it for your templates by calling `{% extends '<template_name>' %}`
+at the top of your template file.
+
+## Object Detail
+
+The most customizable template available to plugin authors is
+`generic/object_detail.html`.
+
+The following blocks are available to you:
+
+- `header`: overloading this block allows for changing the entire top row of
```

Comment:
Maybe use nested lists to reflect the nesting of blocks? I.e.

- header
  - breadcrumbs
    - extra_breadcrumbs
  - buttons
    - extra_buttons
  - masthead
    - title
  - panel_buttons

etc.